src/components/pages/Invite/Invite.tsx

```diff
@@ -0,0 +1,158 @@
+  SEND_INVITE
+}
+
+type ReduxStateProps = {
+  loggedInMember?: Member;
+};
+
+type OwnProps = {};
+
+type InviteProps = ReduxStateProps & OwnProps;
+
+type InviteState = {
+  step: InviteStep;
+  verifiedName?: string;
+  videoUri?: string;
+  videoDownloadUrl?: string;
+};
+
+export class InviteView extends React.Component<InviteProps, InviteState> {
+  inviteToken: string;
+  videoUploadRef: firebase.storage.Reference;
+  dropdown: any;
+
+  constructor(props: InviteProps) {
+    super(props);
+    this.inviteToken = generateToken();
+    this.videoUploadRef = getInviteVideoRef(this.inviteToken);
+    this.state = {
+      step: InviteStep.CAMERA
+    };
+  }
+
+  componentDidMount() {
+    BackHandler.addEventListener("hardwareBackPress", this._handleBackPress);
+  }
+
+  componentWillUnmount() {
+    BackHandler.removeEventListener("hardwareBackPress", this._handleBackPress);
+  }
+
+  _handleBackPress = () => {
+    const index = this.state.step.valueOf();
+    if (index === 0) {
+      // Exit out of Invite flow.
+      return false;
+    } else {
+      const previousStepKey = InviteStep[index - 1];
+      this.setState({
+        step: InviteStep[previousStepKey as keyof typeof InviteStep]
+      });
+      return true;
+    }
+  };
+
+  _verifyVideoUri = () => {
+    const videoUri = this.state.videoUri;
+    if (!videoUri) {
+      this.dropdown.alertWithType(
+        "error",
+        "Error: Can't show video",
+        "Invalid video. Please retake your video."
+      );
+      this.setState({
+        step: InviteStep.CAMERA
+      });
+    }
+    return videoUri;
+  };
+
+  _renderStep() {
+    switch (this.state.step) {
+      case InviteStep.CAMERA: {
+        return (
+          <InviteCamera
+            onVideoRecorded={(videoUri: string) => {
+              this.setState({
+                videoUri: videoUri,
+                step: InviteStep.VIDEO_PREVIEW
+              });
+            }}
+          />
+        );
+      }
+      case InviteStep.VIDEO_PREVIEW: {
+        const videoUri = this._verifyVideoUri();
+        if (!videoUri) {
+          return <React.Fragment />;
+        }
+        return (
+          <VideoPreview
+            videoUri={videoUri}
+            videoUploadRef={this.videoUploadRef}
+            onVideoUploaded={(videoDownloadUrl: string) =>
+              this.setState({
+                videoDownloadUrl: videoDownloadUrl
+                // TODO: step: InviteStep.SEND_INVITE
+              })
+            }
+            onRetakeClicked={() => {
+              this.setState({ step: InviteStep.CAMERA });
+            }}
+            onVideoPlaybackError={(errorMessage: string) => {
+              this.dropdown.alertWithType(
+                "error",
+                "Error: Video Playback",
+                errorMessage
+              );
+              this.setState({
+                step: InviteStep.CAMERA
+              });
+            }}
+          />
+        );
+      }
+      default:
```

Comment:
And also log an error here or below.

Reply:
Same as above, not sure what console.error gains us